modules/programs/dircolors.nix

```diff
@@ -0,0 +1,221 @@
+      ".deb" = mkDefault "01;31";
+      ".rpm" = mkDefault "01;31";
+      ".jar" = mkDefault "01;31";
+      ".war" = mkDefault "01;31";
+      ".ear" = mkDefault "01;31";
+      ".sar" = mkDefault "01;31";
+      ".rar" = mkDefault "01;31";
+      ".alz" = mkDefault "01;31";
+      ".ace" = mkDefault "01;31";
+      ".zoo" = mkDefault "01;31";
+      ".cpio" = mkDefault "01;31";
+      ".7z" = mkDefault "01;31";
+      ".rz" = mkDefault "01;31";
+      ".cab" = mkDefault "01;31";
+      ".wim" = mkDefault "01;31";
+      ".swm" = mkDefault "01;31";
+      ".dwm" = mkDefault "01;31";
+      ".esd" = mkDefault "01;31";
+      ".jpg" = mkDefault "01;35";
+      ".jpeg" = mkDefault "01;35";
+      ".mjpg" = mkDefault "01;35";
+      ".mjpeg" = mkDefault "01;35";
+      ".gif" = mkDefault "01;35";
+      ".bmp" = mkDefault "01;35";
+      ".pbm" = mkDefault "01;35";
+      ".pgm" = mkDefault "01;35";
+      ".ppm" = mkDefault "01;35";
+      ".tga" = mkDefault "01;35";
+      ".xbm" = mkDefault "01;35";
+      ".xpm" = mkDefault "01;35";
+      ".tif" = mkDefault "01;35";
+      ".tiff" = mkDefault "01;35";
+      ".png" = mkDefault "01;35";
+      ".svg" = mkDefault "01;35";
+      ".svgz" = mkDefault "01;35";
+      ".mng" = mkDefault "01;35";
+      ".pcx" = mkDefault "01;35";
+      ".mov" = mkDefault "01;35";
+      ".mpg" = mkDefault "01;35";
+      ".mpeg" = mkDefault "01;35";
+      ".m2v" = mkDefault "01;35";
+      ".mkv" = mkDefault "01;35";
+      ".webm" = mkDefault "01;35";
+      ".ogm" = mkDefault "01;35";
+      ".mp4" = mkDefault "01;35";
+      ".m4v" = mkDefault "01;35";
+      ".mp4v" = mkDefault "01;35";
+      ".vob" = mkDefault "01;35";
+      ".qt" = mkDefault "01;35";
+      ".nuv" = mkDefault "01;35";
+      ".wmv" = mkDefault "01;35";
+      ".asf" = mkDefault "01;35";
+      ".rm" = mkDefault "01;35";
+      ".rmvb" = mkDefault "01;35";
+      ".flc" = mkDefault "01;35";
+      ".avi" = mkDefault "01;35";
+      ".fli" = mkDefault "01;35";
+      ".flv" = mkDefault "01;35";
+      ".gl" = mkDefault "01;35";
+      ".dl" = mkDefault "01;35";
+      ".xcf" = mkDefault "01;35";
+      ".xwd" = mkDefault "01;35";
+      ".yuv" = mkDefault "01;35";
+      ".cgm" = mkDefault "01;35";
+      ".emf" = mkDefault "01;35";
+      ".ogv" = mkDefault "01;35";
+      ".ogx" = mkDefault "01;35";
+      ".aac" = mkDefault "00;36";
+      ".au" = mkDefault "00;36";
+      ".flac" = mkDefault "00;36";
+      ".m4a" = mkDefault "00;36";
+      ".mid" = mkDefault "00;36";
+      ".midi" = mkDefault "00;36";
+      ".mka" = mkDefault "00;36";
+      ".mp3" = mkDefault "00;36";
+      ".mpc" = mkDefault "00;36";
+      ".ogg" = mkDefault "00;36";
+      ".ra" = mkDefault "00;36";
+      ".wav" = mkDefault "00;36";
+      ".oga" = mkDefault "00;36";
+      ".opus" = mkDefault "00;36";
+      ".spx" = mkDefault "00;36";
+      ".xspf" = mkDefault "00;36";
+    };
+
+    home.file.".dir_colors".text = concatStringsSep "\n" ([ ]
```

Comment:
nit: you don’t need the empty array here.

Reply:
True, but I think it provides nicer formatted code. Without the `[ ]`, `nixfmt` gives

```
    home.file.".dir_colors".text = concatStringsSep "\n"
      (optional (cfg.extraConfig != "") cfg.extraConfig
        ++ mapAttrsToList formatLine cfg.settings);
```

Similar RFC 42 modules add that empty array.

Reply:
Either way is fine with me but why create the file at all? Wouldn't it work to have
```
dirColorsFile = pkgs.writeText "dir_colors" (concatStringsSep "\n" …);
```
in the `let` block above and change, e.g.,
```bash
eval $(${pkgs.coreutils}/bin/dircolors -b ~/.dir_colors)
```
to
```bash
eval $(${pkgs.coreutils}/bin/dircolors -b ${dirColorsFile})
```
below?

Reply:
`~/.dir_colors` is the documented location of the `dircolors` config file. I'm not aware of any currently, but I wouldn't be surprised if other applications look for this file. Also, if you use a shell without an `enableShellIntegration` option, you will need that file to use `dircolors`. I suppose I could add an option like `linkConfig` to add the config file to your home. If so, I'm not sure if the config file should  be included in your home by default or not.